blocks/bricks/sequence_generators.py

```diff
@@ -257,6 +266,13 @@ def cost_matrix(self, application_call, outputs, mask=None, **kwargs):
             mask=mask, return_initial_states=True, as_dict=True,
             **dict_union(inputs, states, contexts))
 
+        # Run the language model
+        if self.language_model:
+            lm_arguments = self.language_model.apply(
+                outputs, mask=mask, as_dict=True)
```

Comment:
I see what you mean and you are right, but wouldn't be sufficient to shift `lm_arguments` just like we do it with `states`?